packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -228,6 +227,21 @@ export function checkListMembershipInput(
     }
   }
 
+  // All lists of tuples should be arity homogeneous.
+  for (const [listName, validValueList] of Object.entries(membershipLists)) {
+    if (Array.isArray(validValueList[0])) {
```

Comment:
Doesn't this if need a corresponding else?  If the first value is not a tuple, the remaining entries should also not be tuples.

Reply:
This will be caught by the type system since `validValueList: PODValue[] | PODValue[][]`. I'll comment on that.

Reply:
True, though I don't really trust the type system in the "check" phase, since it's used immediately after deserialization.

Reply:
Fair enough. I'll put that check in then.